Solver/elastic_law.py

```diff
@@ -0,0 +1,310 @@
+        if c2 != 0.0:
+            dw_int += inner(c2 * div(grad(u)), grad(div(v)))
+        if c3 != 0.0:
+            dw_int += inner(c3 * grad(div(u)), grad(div(v)))
+        if c4 != 0.0:
+            for i in range(self.dim):
+                for j in range(self.dim):
+                    for k in range(self.dim):
+                        mat = inner(c4 * Dx(Dx(u[i], j), k),
+                                    Dx(Dx(v[i], j), k))
+                        if isinstance(mat, list):
+                            dw_int += mat
+                        else:
+                            dw_int += [mat]
+        if c5 != 0.0:
+            for i in range(self.dim):
+                for j in range(self.dim):
+                    for k in range(self.dim):
+                        mat = inner(c5 * Dx(Dx(u[j], i), k),
+                                    Dx(Dx(v[i], j), k))
+                        if isinstance(mat, list):
+                            dw_int += mat
+                        else:
+                            dw_int += [mat]
+        dw_int += inner(mu * grad(u), grad(v))
+
+        for i in range(self.dim):
+            for j in range(self.dim):
+                mat = inner(mu * Dx(u[i], j), Dx(v[j], i))
+                if isinstance(mat, list):
+                    dw_int += mat
+                else:
+                    dw_int += [mat]
+        dw_int += inner(lmbda * div(u), div(v))
+
+        return dw_int
+
+    def compute_body_forces(self, u):
+        assert isinstance(u, tuple)
+        assert hasattr(self, "material_parameters")
+        assert len(self.material_parameters) == self.nb_material_parameters
+
+        dim = len(u)
+        lmbda, mu, c1, c2, c3, c4, c5 = self.material_parameters
+
+        x, y, z = sp.symbols("x,y,z")
+        coord = [x, y, z]
+
+        Divergence = 0.
+        for i in range(dim):
+            Divergence += u[i].diff(coord[i])
+
+        Laplace = np.zeros(dim, dtype=sp.Expr)
+        for i in range(dim):
+            for j in range(dim):
+                Laplace[i] += u[i].diff(coord[j], 2)
+
+        GradDiv = np.empty(dim, dtype=sp.Expr)
+        for i in range(dim):
+            GradDiv[i] = Divergence.diff(coord[i])
+
+        DoubleLaplace = np.zeros(dim, dtype=sp.Expr)
+        for i in range(dim):
+            for j in range(dim):
+                DoubleLaplace[i] += Laplace[i].diff(coord[j], 2)
+
+        DivDivGrad = 0.
+        for i in range(dim):
+            DivDivGrad += Laplace[i].diff(coord[i])
+
+        GradDivDivGrad = np.empty(dim, dtype=sp.Expr)
+        for i in range(dim):
+            GradDivDivGrad[i] = DivDivGrad.diff(coord[i])
+
+        body_forces = (c1 + c4) * DoubleLaplace + \
+            (c2 + c3 + c5) * GradDivDivGrad - \
+            (lmbda + mu) * GradDiv - mu * Laplace
+
+        return tuple(body_forces)
+
+    def set_material_parameters(self, material_parameters):
+        self.material_parameters = material_parameters
```

Comment:
Here `material_parameters` should become a property with a setter and getter method. By the way, there is no consistency check regarding the number of material parameters and the type of the value.